### DISCUSSION TITLE
`@mantine/notifications` Allow global notifications auto close customization w/ call context

### DISCUSSION THREAD

**Discussion by dillonstreator** [mantinedev/mantine#5383] [Feature requests]
I'd like to propose extending the `autoClose` functionality by allowing a callback to be specified (in addition to the existing `number | false`) that receives the `NotificationData` and returns `number | false`. This enables creating a "global" default ruleset/config on the `Notifications` component.
This global config should be overridable at the time of calling `showNotification` by specifying `autoClose`.

Here is an example of what the implementation could look like https://github.com/mantinedev/mantine/pull/5329

**Comment by arvind0598** [comment]
@rtivital do you have any opinions on this? My user-base happens to be slow readers, and this would help out quite a bit.

I'm willing to continue the work on #5329 if there are any issues with the PR or if @dillonstreator is unavailable.
